src/alfasim_score/common.py

```diff
@@ -58,6 +67,114 @@ def _missing_(cls, value):  # type: ignore
         raise ValueError(f"{value} is not a valid {cls.__name__}. Valid types: {available}")
 
 
+class FluidModelType(str, Enum):
+    PVT = "PVT"
+    ZAMORA = "Zamora"
+
+
+class AnnulusModeType(str, Enum):
+    UNDISTURBED = "Undisturbed"
+    DRILLING = "Drilling"
+
+
+@dataclass
+class FluidModelPvt:
+    name: str
+
+    def to_dict(self) -> Dict[str, Union[str, Scalar]]:
+        """Convert data to dict in order to write data to the alfacase."""
+        return {
+            "name": self.name,
+            "fluid_type": FluidModelType.PVT.value,
+        }
+
+
+@dataclass
+class SolidMechanicalProperties:
+    name: str
+    young_modulus: Scalar
+    poisson_ratio: Scalar
+    thermal_expansion_coefficient: Scalar
+
+    def to_dict(self) -> Dict[str, Any]:
+        """Convert data to dict in order to write data to the alfacase."""
+        return asdict(self)
+
+
+@dataclass
+class AnnulusTable:
+    fluid_names: List[str] = field(default_factory=lambda: [])
+    fluid_ids: List[float] = field(default_factory=lambda: [])
+    initial_depths: Array = field(default_factory=lambda: Array([], LENGTH_UNIT))
+    final_depths: Array = field(default_factory=lambda: Array([], LENGTH_UNIT))
+
+    def to_dict(self, annulus_type: str) -> Dict[str, Any]:
+        """Convert data to dict in order to write data to the alfacase."""
+        columns = {
+            f"fluid_id_{annulus_type}": self.fluid_ids,
+            f"initial_depth_{annulus_type}": self.initial_depths,
+            f"final_depth_{annulus_type}": self.final_depths,
+        }
+        return {"columns": columns}
+
+
+@dataclass
+class Annulus:
+    is_active: bool = False
+    mode_type: AnnulusModeType = AnnulusModeType.UNDISTURBED
+    initial_top_pressure: Scalar = Scalar(0.0, PRESSURE_UNIT)
+    is_open_seabed: bool = False
+    annulus_table: AnnulusTable = field(default_factory=lambda: AnnulusTable())
+    has_fluid_return: bool = False
+    initial_leakoff: Scalar = Scalar(0.0, VOLUME_UNIT)
+    has_relief_pressure: bool = False
+    relief_pressure: Scalar = Scalar(0.0, PRESSURE_UNIT)
+    relief_position: Scalar = Scalar(0.0, LENGTH_UNIT)
+
+    def to_dict(self, annulus_type: str) -> Dict[str, Any]:
+        """Convert data to dict in order to write data to the alfacase."""
+        # TODO PWPA-2152: make sure all names match (plugin and converter) then remove this mapped names
+        plugin_key_names = {
+            "is_active": "is_active",
+            "mode_type": "mode_type",
+            "initial_top_pressure": "initial_top_pressure",
+            "is_open_seabed": "open_sea",
+            "annulus_table": "annulus_table",
+            "has_fluid_return": "fluid_return",
+            "initial_leakoff": "initial_leakoff",
+            "has_relief_pressure": "relief_pressure_check",
+            "relief_pressure": "pressure_relief",
+            "relief_position": "relief_position",
```

Comment:
Some things must change here, since I've followed your suggestions on PWPA-2113.

Reply:
Also add the new temperature vs depth table.

Reply:
ok, done in the next PR.